File: nodejs/dynamoClient.js
const {DynamoDB} = require("aws-sdk")

const client = new DynamoDB()

function factory(TableName) {

  return {
    add(Item, callback) {
      Item.id = `unique-identifier-${Item.name.toLowerCase()}`

      return new Promise((resolve, reject) =>
        client.putItem({TableName, Item}, (error, response) =>
          error ? reject(error, response) : resolve(response)))
    }
  }
}

module.exports = factory

File: python/dynamoClient.py
import boto3


class Factory:
    def __init__(self, tablename, region):
        self.tablename = tablename
        self.region = region

    def add(self, Item):
        try:
            client = boto3.resource('dynamodb', region_name=self.region)
            return client.Table(self.tablename).put_item(Item=Item)
        except Exception as e:
            raise e
